ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/MultiTenancyPrivacyController.java

```diff
@@ -72,8 +72,19 @@ public String sendTransaction(
       verifyPrivacyGroupContainsEnclavePublicKey(
           privateTransaction.getPrivacyGroupId().get().toBase64String(), enclavePublicKey);
     }
-    return privacyController.sendTransaction(
-        privateTransaction, enclavePublicKey, maybePrivacyGroup);
+
+    final String transactionResult =
+        privacyController.sendTransaction(privateTransaction, enclavePublicKey, maybePrivacyGroup);
+
+    if (isGroupRemovalTransaction(privateTransaction)) {
```

Comment:
We can't do this here, it is too early in the private tx lifecycle. We need to send the event when the transaction is processed as part of the block import.
Basically, only after we know the tx has been successfully processed we can send the event.